packages/zoe/src/contracts/operaConcertTicket.js

```diff
@@ -0,0 +1,113 @@
+import { makeZoeHelpers, defaultAcceptanceMsg } from './helpers/zoeHelpers';
+
+import produceIssuer from '@agoric/ertp';
+
+/*
+  Roles in the arrangement:
+  - Contract creator: describes the contract with:
+    - number of seats, show, date/time of start
+    - expected (ERTP) amount per ticket (we assume all tickets cost the same)
+  - Smart Contract: 
+    - mints the tickets
+    - provides the seats
+  - Auditorium (unique contract seat, usually taken by the contract creator): the person hosting 
+  the Opera show, selling the tickets and getting the payment back
+  - Ticket buyers (contract seat created on demand): 
+    - can see the available opera show seats
+    - can consult the terms
+    - can redeem the zoe invite with the proper payment to get the ticket back
+*/
+
+export const makeContract = harden(async zoe => {
+
+  const zoeService = zoe.getZoeService();
+
+  const {
+    rejectOffer,
+    checkIfProposal,
+    swap,
+    canTradeWith,
+    getActiveOffers,
+    assertKeywords,
+  } = makeZoeHelpers(zoe);
+
+  // Create the internal ticket mint
+  const { issuer, mint, amountMath } = produceIssuer('Opera tickets', 'set');
+  await zoe.addNewIssuer(issuer, 'Auditorium');
+  
+  const { terms: { show, start, count, expectedAmountPerTicket } } = zoe.getInstanceRecord();
+  
+  const ticketDescriptionObjects = Array(count)
+    .fill()
+    .map((_, i) => harden({ show, start, number: i + 1 }));
+
+  const availableTicketAmountsByTicketNumber = new Map(
+    ticketDescriptionObjects.map(description => [description.number, amountMath.make(harden([description]))])
+  )
+
+  // Create the offers in Zoe for the tickets
+  const ticketPaymentInviteHandles = new Map(await Promise.all(
+    [...availableTicketAmountsByTicketNumber.entries()].map(([ticketNumber, amount]) => {
+      // create an Zoe invite internally...
+      const {invite, inviteHandle} = zoe.makeInvite()
+      // ...and redeem it right away
+      return zoeService.redeem(
```

Comment:
The approach i've taken is that since the mint is internal to the smart contract, the smart contract needs to create its own invitations to send the offers to Zoe. Thesmart  contract immediatly redeems the invites (and ignores the seat since it's irrelevant from the smart contract perspective)

@katelynsills I'm interested in your opinion on this approach
I didn't see an easier way. I wonder whether there could be a zoe contract facet method to make this easier (but is it really a relevant use case given the mint being internal is a temporary decision?)

Reply:
Maybe that's where `makeEmptyOffer` is relevant